### ISSUE TITLE
Squiz Unreachable code detection is not working properly with a closure inside a case

### ISSUE THREAD

**Issue by VasekPurchart** [squizlabs/PHP_CodeSniffer#1007]
Hi, 

I've found out, that there is a special case where the `Squiz.PHP.NonExecutableCode.Unreachable` reports a false positive if there is a call with a closure inside a `case` statement. For this example:

``` php
<?php

switch (true) {
	case 1:
		return foo(
			function () {
				$foo = $bar;  // when this is removed it works ok
				return false; // from here on it reports unreachable
			}
		);
}
```
it reports:

```
 8 | WARNING | Code after RETURN statement cannot be executed
   |         | (Squiz.PHP.NonExecutableCode.Unreachable)
```

I've tried several other combinations such as:

``` php
<?php

function f()
{
	return foo(
		function () {
			$foo = $bar;
			return false;
		}
	);
}

switch (true) {
	case 1:
		return foo(
			function () {
				return false;
			}
		);
}
```

and both these examples are working correctly. The most interesting is the last example, because the only difference to the broken one is the missing `$foo = $bar;` line.

---

Tested on `2.6`

**Comment by gsherwood** [comment]
Fixed now. Thanks for reporting it.

**Comment by VasekPurchart** [comment]
Thanks for the fix :)
